### PR TITLE
Use pytest as test runner

### PR DESCRIPTION
**Pull request by smithdc1**
* Used pytest to run tests within tox environments

* Removed test runner within setup.py

@bittner is this what you were thinking of to refactor the test runner out of setup.py?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import os
 
 try:
-    from setuptools import setup, Command
+    from setuptools import setup
 except ImportError:
-    from distutils.core import setup, Command
-
-os.environ['DJANGO_SETTINGS_MODULE'] = 'analytical.tests.settings'
+    from distutils.core import setup
```
Comment (bittner) [inline setup.py:6]: I think this try-except is unneeded in modern Python 3.

File: setup.cfg [hunk 2]
```diff
@@ -5,3 +5,6 @@ all_files  = 1
 
 [upload_sphinx]
 upload-dir = build/docs/html
+
+[tool:pytest]
+DJANGO_SETTINGS_MODULE = analytical.tests.settings
```
Comment (bittner) [inline setup.cfg:10]: Newline missing at end of line.

File: tox.ini [hunk 3]
```diff
@@ -10,11 +10,13 @@ envlist =
 description = Unit tests
 deps =
     coverage
+    pytest
```
Comment (bittner) [inline tox.ini:13]: You can probably omit that as a dependency when you specify `pytest-django`.

**Review by bittner — CHANGES_REQUESTED** [review]
Not bad! That looks like a simple but effective change. Nice!  :+1: 

There are a few warnings in the `tox` runs, which however will be fix by PR #170, so we don't need to worry.

I made a few minor comments.

File: setup.py [hunk 4]
```diff
@@ -17,42 +15,9 @@
...
-    user_options = []
-
-    def initialize_options(self):
-        pass
-
-    def finalize_options(self):
-        pass
-
-    def run(self):
-        from analytical.tests.utils import run_tests
```
Comment (smithdc1) [inline setup.py:31]: I'll remove this function as well.

**Comment by bittner** [comment]
Can you try to see if it's possible to also [remove the tests module](https://github.com/jazzband/django-analytical/blob/master/setup.py#L70-L71) from the packaging setup already?

```patch
--- a/setup.py
+++ b/setup.py
     packages=[
         'analytical',
         'analytical.templatetags',
-        'analytical.tests',
-        'analytical.tests.templatetags',
     ],
```

File: setup.py [hunk 5]
```diff
@@ -17,42 +15,9 @@
...
-        "analytical.tests.settings"
-    )
-    django.setup()
-except ImportError:
-    print(
-        "Could not import django. "
-        "This is fine, unless you intend to run unit tests."
-    )
-
 import analytical as package  # noqa
```
Comment (bittner) [inline setup.py:21]: Move this import statement up to its regular place, where all the others are, then you'll be able to remove the `# noqa` safely. Now this should be possible.

File: setup.py [hunk 6]
```diff
@@ -17,42 +15,9 @@
     pass
```
Comment (bittner) [inline setup.py:15]: The `cmdclass` variable and this entire block can also go away if you move the docs generation into `tox.ini` ([see example](https://github.com/behave/behave-django/blob/master/tox.ini#L45-L50)). :rocket:

File: setup.py [hunk 7]
```diff
@@ -17,42 +15,9 @@
...
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
-
-try:
-    import django
-    os.environ.setdefault(
-        "DJANGO_SETTINGS_MODULE",
-        "analytical.tests.settings"
-    )
-    django.setup()
```
Comment (bittner) [inline setup.py:49]: Great! Having this removed, we should now be able to declare #117 to be fixed! :rocket:

**Comment by smithdc1** [comment]
Hi @bittner -- I've addressed your comments here. +7 / -72 seems like a nice win here. :rocket:

**Review by bittner — APPROVED** [review]
Sweet! I think we're ready to merge. :rocket: 

Good job! :+1:

**Pull request merged**